enacts/wat_bal/agronomy.py

```diff
@@ -214,3 +226,153 @@ def soil_plant_water_balance(
+        indicating the upper limit (inclusive) to belong to an API category
+    api_poly : DataArray, optional
+        polynomial coefficients that must depend on a dimension called powers
+        of size 3 (the 3 powers of a polynomial of order 2),
+        and on a dimension api_cat of size one more than `api_thresh` 's size.
+        The polynomial used to compute the `runoff` is picked according to the categories
+        defined by the thresholds.
+    time_dim : str, optional
+        daily time dimension of `daily_rain` (default `time_dim` ="T").
+        
+    Returns
+    -------
+    runoff : DataArray
+        daily Runoff.
+
+    See Also
+    --------
+    api_sum
+    
+    Notes
+    -----
+    The default `api_thresh` is
+
+    >>> xr.DataArray([6.3, 19, 31.7, 44.4, 57.1, 69.8], dims=["api_cat"])
+
+    and the default `api_poly` is
+
+    >>> xr.DataArray(
+    >>>     [
+    >>>         [0.858, 0.0895, 0.0028],
+    >>>         [-1.14, 0.042, 0.0026],
+    >>>         [-2.34, 0.12, 0.0026],
+    >>>         [-2.36, 0.19, 0.0026],
+    >>>         [-2.78, 0.25, 0.0026],
+    >>>         [-3.17, 0.32, 0.0024],
+    >>>         [-4.21, 0.438, 0.0018],
+    >>>     ],
+    >>>     dims=["api_cat", "powers"],
+    >>> )
+
+    Which means dor instance that, if rain is greater or equal to 12.5
+    and API is 18, then runoff is
+    
+    -1.14 + 0.042*x 0.0026*(x**2)
+    
+    where x is daily rain.
+    """
+    if api_thresh is None:
+       api_thresh = DEFAULT_API_THRESHOLD.copy(deep=True)
+    if api_poly is None:
+        api_poly = DEFAULT_API_POLYNOMIALS.copy(deep=True)
+    
+    # Compute API
+    api = daily_rain.rolling(**{time_dim:7}).reduce(api_sum).dropna(time_dim)
+    runoff = xr.dot(
+        xr.dot(
+            api_poly,
+            xr.concat(
+                [np.power(daily_rain, 0), daily_rain, np.square(daily_rain)],
+                dim="powers",
+            ).where(daily_rain > no_runoff, 0),
+        ),
+        xr.concat(
+            [api <= api_thresh, ~np.isnan(api)],
+            dim="api_cat",
+        ).cumsum(dim="api_cat").where(lambda x: x <= 1, other=0),
+        dims="api_cat",
+    ).clip(min=0).rename("runoff")
+    runoff.attrs = dict(description="Runoff", units="mm")
+    return runoff
+
+
+def api_runoff(
+    daily_rain,
+    api,
+    no_runoff=12.5,
+    api_thresh=None,
+    api_poly=None,
+):
+    if api_thresh is None:
+        api_thresh = DEFAULT_API_THRESHOLD
+    if api_poly is None:
+        api_poly = DEFAULT_API_POLYNOMIALS
+    print([daily_rain <= no_runoff]
+            + [api <= api_v for k, api_v in enumerate(api_thresh)])
+    func = lambda dr, i, nr:  np.piecewise(
+        dr,
+        [dr <= nr]
+            + [i <= api_v for k, api_v in enumerate(api_thresh)],
+        [0]
+            + [lambda x: np.polynomial.polynomial.polyval(x, api_coeff)
+                for k, api_coeff in enumerate(api_poly)],
```

Comment:
I think this can be written more simply using `Polynomial` instead of `polyval`.
```
[np.polynomial.polynomial.Polynomial(api_coeff) for api_coeff in api_poly]
```
because an instance of the `Polynomial` class can be called as a function:
```
>>> p = np.polynomial.polynomial.Polynomial([1, 1])
>>> p
Polynomial([1., 1.], domain=[-1,  1], window=[-1,  1])
>>> p(1)
2.0
>>> p(2)
3.0
>>> p(3)
4.0
```